season-scraper/leagues/americanHockeyLeague.js

```diff
@@ -0,0 +1,99 @@
+//ahl scraper for npm module
+
+const utils = require('../../utils')
+
+function getSeasons(prospect) {
+  const unFilteredSeasons = [...prospect.careerStats[0].sections[0].data, ...prospect.careerStats[0].sections[1].data].map((season) => season.row)
+  
+  //The raw data has a season called 'total' in both arrays.  Removing it for now
+  const filteredSeasons = unFilteredSeasons.filter((season) => season.season_name != 'Total')
+  return filteredSeasons.map((season) => getFormattedSeason(season))
+
+}
+
+function getInfo(info) {
+  const age = utils.calculation.ageOnDate(info.birthDate)
+
+  return {
+    number: info.jerseyNumber,
+    position: info.position,
+    age: age
+  }
+}
+//checks if season is the playoffs and returns all associated properties
+function getSeasonTypeProperties (seasonName) {
+  const startYear = parseInt(seasonName.substr(0,4))
```

Comment:
I think you might be able to do something simpler with regex:

```
const [startYear, endYear] = seasonName.match(/(\d{4})/)
const seasonType = +startYear !== +endYear - 1 ? "Playoffs" : "Regular"

return { startYear: +startYear, endYear: +endYear, seasonType }
```

Note that I did not test the regex, need a sec to think about it

Reply:
You're just matching 4 digits for the regex? 

Only problem I see is I think you're returning end year as undefined here.

Reply:
True, I fixed that. Also needed the `/g` on the regex to get both years

Reply:
Yeah and its probably pretty trivial to do an edge case for years that are expressed in 2 digits.  Something like if no matches return 2 digit matches and add 2000 to them.  Which likely means you can use it for every script.